src/drivers/dw/dma.c

```diff
@@ -512,7 +512,7 @@ static int dw_dma_start(struct dma *dma, int channel)
 			start_offset = PLATFORM_TIMER_START_OFFSET;
 
 		/* activate timer for timer driven scheduling */
-		work_schedule_default(&chan->dma_ch_work,
+		ll_schedule_default(&chan->dma_ch_work,
```

Comment:
The scheduler APIs will all start with "schedule_" and be consistent whether we use EDF or work scheduler. A flag can be used by caller to select, alternatively the scheduling type can be stored in the work/task structure.

Reply:
It's ok in PR's HEAD